kedro-datasets/RELEASE.md

```diff
@@ -10,6 +10,7 @@
 | `langchain.OpenAIEmbeddingsDataset` | A dataset for loading a OpenAIEmbeddings langchain model. | `kedro_datasets_experimental.langchain` |
 | `langchain.ChatOpenAIDataset`       | A dataset for loading a ChatOpenAI langchain model.       | `kedro_datasets_experimental.langchain` |
 * Extended preview feature to `yaml.YAMLDataset`.
+* Added new `CSVDataset` to [dask](https://www.dask.org/) module.
```

Comment:
Nit: we usually add new datasets in a table in the release notes, like above for the experimental dataset.